shell/platform/darwin/macos/framework/Source/FlutterEngineTest.mm

```diff
@@ -39,6 +39,23 @@ @interface FlutterEngine (Test)
   EXPECT_TRUE(engine.running);
 }
 
+TEST_F(FlutterEngineTest, HasNonNullExecutableName) {
+  // Launch the test entrypoint.
+  FlutterEngine* engine = GetFlutterEngine();
+  std::string executable_name = [[engine executableName] UTF8String];
+  EXPECT_TRUE([engine runWithEntrypoint:@"executableNameNotNull"]);
+
+  // Block until notified by the Dart test of the value of Platform.executable.
+  fml::AutoResetWaitableEvent latch;
+  AddNativeCallback("NotifyStringValue", CREATE_NATIVE_ENTRY([&](Dart_NativeArguments args) {
+                      const auto dart_string = tonic::DartConverter<std::string>::FromDart(
+                          Dart_GetNativeArgument(args, 0));
+                      EXPECT_EQ(executable_name, dart_string);
```

Comment:
Is the value of `executable_name` well known? Would it be worthwhile to assert on that as well? I don't feel strongly about this, feel free to skip

Reply:
Good point!

The path itself is subject to our infra and the name of the build target, so we can't necessarily guarantee the exact value in a non-fragile way, but that's a good point -- there are certain assertions we can safely make: at a minimum, we know that it's not the empty string and we should probably assert that.